### DISCUSSION TITLE
Do we need to allow changing of range of symbols in plot.

### DISCUSSION THREAD

**Discussion by shishir-11** [sympy/sympy#26493] [Q&A]
Consider this [issue](https://github.com/sympy/sympy/issues/26377) .Although it is clearly mentioned in the documentation about the default range of the symbol , would it be better to change the default range to somewhere in the axis limit if axis limit and symbol range do not coincide.

**Comment by sylee957** [comment]
Even though it may be possible to write some code to infer the default range, if other user option like `xlim` is given, 
I'm not certain about if we should implement that, unless the implementation looks clean and concise and doesn't break some other implementation. 
(I'd not be pushing such ideas to the community because in my experiences, the actual implementations doesn't look very clean, and generally, I don't think that the feature is necessary to achieve any work)

Anyway, my advice is that you should stay away from `xlim` and should always use default range
such as `plot(f, (x, -10, 10))` because it is very trivial to think of, 
and it makes the code more robust and explicit. 
You should always know that plot is drawn from `-10` to `10` (or something else you explicitly specify), however, in other ways, if some SymPy version changes the way default range is inferred, you get surprising results.
